x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/action/EsqlExecutionInfo.java

```diff
@@ -240,6 +252,7 @@ public Iterator<? extends ToXContent> toXContentChunked(ToXContent.Params params
             b.field(SKIPPED_FIELD.getPreferredName(), getClusterStateCount(Cluster.Status.SKIPPED));
             b.field(PARTIAL_FIELD.getPreferredName(), getClusterStateCount(Cluster.Status.PARTIAL));
             b.field(FAILED_FIELD.getPreferredName(), getClusterStateCount(Cluster.Status.FAILED));
+            b.field(IS_PARTIAL_FIELD.getPreferredName(), isPartial);
```

Comment:
I'm not sure we should add this here yet. We may want to leave it out until we sync with the is_partial work that @luigidellaquila is doing?  

For _search the is_partial setting in the response is only present in the async-search response. Here it would also be present for sync searches, right?  So that would need to be discussed and agreed with the larger team (and possible Search Foundations), as well.

For right now, my vote is that we we should just mark the cluster as partial status.

Reply:
We could exclude it for sync queries, maybe? Sync query can't be partial for now.